tools/update-typescript-tests.js

```diff
@@ -26,50 +21,60 @@ var shelljs = require("shelljs"),
 
 /**
  * Returns a raw copy of the given AST
- * @param  {object} ast the AST object
- * @returns {object}     copy of the AST object
+ * @param  {Object} ast the AST object
+ * @returns {Object}     copy of the AST object
  */
 function getRaw(ast) {
     return JSON.parse(JSON.stringify(ast));
 }
 
-
+/**
+ * Returns the expected AST
+ * @param  {string} code the source code
+ * @param {Object} config the parser config
+ * @returns {Object}     the expected AST object
+ */
 function getExpectedResult(code, config) {
     try {
         return tester.getRaw(parser.parse(code, config));
     } catch (ex) {
-        var raw = getRaw(ex);
+        const raw = getRaw(ex);
         raw.message = ex.message;
         return raw;
     }
 }
 
+/**
+ * Returns an array of relevant filenames within the directory
+ * @param {string} directory path of the directory to scan
+ * @returns {string[]} list of relevant filenames
+ */
 function getTestFilenames(directory) {
-    return shelljs.find(directory).filter(function(filename) {
-        return filename.indexOf(".src.ts") > -1;
-    }).map(function(filename) {
-        return filename.substring(directory.length - 1, filename.length - 7);  // strip off ".src.ts"
-    });
+    return shelljs.find(directory).filter(filename => filename.indexOf(".src.ts") > -1).map(filename => filename.substring(directory.length - 1, filename.length - 7)  // strip off ".src.ts"
+    );
 }
 
+/**
+ * Writes the given AST to the given filename
+ * @param {*} result the AST to write to a file
+ * @param {*} testResultFilename name of the file to write to
+ * @returns {void}
+ */
 function outputResult(result, testResultFilename) {
-    shelljs.echo("module.exports = " + JSON.stringify(result, null, "    ") + ";").to(testResultFilename);
+    shelljs.echo(`module.exports = ${JSON.stringify(result, null, "    ")};`).to(testResultFilename);
```

Comment:
I thought about using this tool but sometimes it will generate JSON objects that are inconsistent or change for no reason. By adding the deep equal assertion it does prevent updating every file but maybe we should make the stringifyed JSON output consistent. We can use a tool like this: https://github.com/substack/json-stable-stringify to sort the output keys. This could be done later just a thought.